<launch>

	<!-- Include launch files -->
	<!-- miminal launch -->
	<include file="$(find turtlebot_bringup)/launch/minimal.launch"/>


   	<!-- teleop -->
    	<!-- TO DO::: ADD ARGUMENT TO CHOOSE BETWEEN JOYSTICK OR KEYBOARD TELEOP -->
    <include file="$(find turtlebot_teleop)/launch/keyboard_teleop.launch"/>
	
   	<!-- 3D visualization -->
	<include file="$(find turtlebot_bringup)/launch/3dsensor.launch"/>
 

   	<!-- rviz -->
	<include file="$(find turtlebot_rviz_launchers)/launch/view_robot.launch"/>

	<!-- run nodes -->
	<!-- motion controller node -->
	<!--include file="$(find my_tutorial)/launch/motion_controller.launch"/-->
	<node name="motion_model" pkg="my_tutorial" type="motion_model.py" required="true" output="screen" launch-prefix="xterm -e"/>
	
	<!-- measuremetn processing node -->
	<!--include file="$(find my_tutorial)/launch/measurement_processing.launch"/-->
	<node name="meas_update" pkg="my_tutorial" type="meas_update.py" required="true" output="screen" launch-prefix="xterm -e"/>
	
	<!-- reference provider -->
	<!--include file="$(find my_tutorial)/launch/reference_provider.launch"/-->
	<node name="reference_provider" pkg="my_tutorial" type="reference_provider.py" required="true" output="screen" launch-prefix="xterm -e"/>

</launch>
